src/pages/admin/Event/EventStats/MassUpdateModal.js

```diff
@@ -0,0 +1,124 @@
+import React, {
+  useState
+} from "react";
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  TextField,
+  Select,
+  MenuItem,
+  DialogActions,
+  Button
+} from "@material-ui/core";
+import {
+  APPLICATION_STATUS
+} from "constants/_constants/eventStatsStatusFields";
+import {
+  fetchBackend
+} from "utils";
+
+const MassUpdateModal = ({
+  open,
+  onClose,
+  eventID,
+  eventYear,
+  refreshTable
+}) => {
+  const [emailList, setEmailList] = useState("");
+  const [selectedStatus, setSelectedStatus] = useState("");
+
+  const fetchFnameByEmail = async (email) => {
+    try {
+      const response = await fetchBackend(`/members/${email}/fname`, "GET");
```

Comment:
I noticed that there was already an updateHelper, so I wanted to use that, but it required the user's fname. 

I think you're right. My current implementation wouldn't work for non-users then, since they aren't in the members table. I'm guessing I can't use the existing updateHelper function then?

Should I work on recreating the update status function so that it doesn't require fname? Or do you think there's another way to get fname without using the members table?

Reply:
nvm resolved dennis showed me how to grab fname directly from row